src/include/storage/page/extendible_htable_header_page.h

```diff
@@ -0,0 +1,57 @@
+//===----------------------------------------------------------------------===//
+//
+//                         BusTub
+//
+// extendible_htable_header_page.h
+//
+// Identification: src/include/storage/page/extendible_htable_header_page.h
+//
+// Copyright (c) 2015-2023, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+/**
+ * Header page format:
+ *  ---------------------------------------------------
+ * | DirectoryPageIds(2048) | MaxSize (4) | Free(2044)
+ *  ---------------------------------------------------
+ */
+
+#pragma once
+
+#include <cstdlib>
+#include "common/config.h"
+#include "hash_table_page_defs.h"
+
+namespace bustub {
+#define HTABLE_HEADER_PAGE ExtendibleHTableHeaderPage
+#define HTABLE_HEADER_PAGE_METADATA_SIZE 4
+#define HTABLE_HEADER_ARRAY_SIZE 512
+
+INDEX_TEMPLATE_ARGUMENTS
+class ExtendibleHTableHeaderPage {
+ public:
+  // Delete all constructor / destructor to ensure memory safety
+  ExtendibleHTableHeaderPage() = delete;
+  ExtendibleHTableHeaderPage(const ExtendibleHTableHeaderPage &other) = delete;
+
+  /**
+   * After creating a new header page from buffer pool, must call initialize
+   * method to set default values
+   * @param max_size Max size of the array in the header page
+   */
+  void Init(int max_size = HTABLE_HEADER_ARRAY_SIZE);
+
+  auto HashToDirectoryPageId(uint32_t hash) -> page_id_t;
+
+  /**
+   * Prints the header's occupancy information
+   */
+  void PrintHeader();
+  
+ private:
+	page_id_t directory_page_ids_[HTABLE_HEADER_ARRAY_SIZE];
```

Comment:
nit: indentation and maybe ordering? we can put max_size_ before the array like the other page layouts

otherwise LGTM!